### PR TITLE
[BUGFIX] Make ExpectationContext optional and remove when null to ensure backwards compatability

### PR DESCRIPTION
**Pull request by cdkini**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- Hotfix PR for incompatible change made to ExpectationSuite schema

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/core/expectation_configuration.py
Comment (cdkini) [inline great_expectations/core/expectation_configuration.py:1390]: This value is not mandatory so we should make it optional. If it is null, let's remove it before converting to JSON. I believe this approach should be standardized for newer attributes.

File: great_expectations/core/expectation_configuration.py
Comment (cdkini) [inline great_expectations/core/expectation_configuration.py:1291]: @post_dump cleans this completely so we need to verify the key's existence before conversion.

Could also simply do:
```python
for attr in myself.__dict__:
    myself[attr] = convert_to_json_serializeable(attr)
```
Don't know if that has any implications so kept it simple.

File: great_expectations/core/expectation_configuration.py
Comment (cdkini) [inline great_expectations/core/expectation_configuration.py:972]: If you'd like to use the new functionality, you must explicitly pass it in. Is this a fair assumption?

Not sure if this violates any expected behavior that Cloud relies on.

File: great_expectations/core/expectation_configuration.py
Comment (cdkini) [inline great_expectations/core/expectation_configuration.py:1049]: No longer a guaranteed value; this tracks with how we treat ge_cloud_id

**Review by alexsherstinsky — APPROVED** [review]
LGTM (once all checks pass -- I had to rerun the usage statistics test; probably an intermittent issue with the Lambda).

File: great_expectations/core/expectation_configuration.py
Comment (talagluck) [reply great_expectations/core/expectation_configuration.py:1390]: Love this - I think it will save us a lot of pain moving forward. Let's make sure to include this in the discussion on code standards.

**Review by talagluck — APPROVED** [review]
LGTM! I really like this approach. I think we should be good to merge, pending a review from someone on Cloud.

**Pull request merged**
